include/wx/msw/winundef.h

```diff
@@ -15,30 +15,52 @@
  */
 
 // ----------------------------------------------------------------------------
-// windows.h #defines the following identifiers which are also used in wxWin so
-// we replace these symbols with the corresponding inline functions and
-// undefine the macro.
+// windows.h and windowsx.h #define the following identifiers which are also
+// used by wxWidgets so we replace these symbols with the corresponding inline
+// functions and undefine the macro.
 //
 // This looks quite ugly here but allows us to write clear (and correct!) code
 // elsewhere because the functions, unlike the macros, respect the scope.
 // ----------------------------------------------------------------------------
 
-// CreateDialog
+#if defined(UNICODE) && (defined(WXBUILDING) || !defined(wxUNICODE_SET_BY_WXWIDGETS))
+    // inline the WIDE Win32 API functions when building wxWidgets or when
+    // UNICODE is not defined by wxWidgets (i.e. the user has defined UNICODE).
+    #define wxUSE_UNICODE_WINDOWS_H 1
+
+    #define wxLPCTSTR LPCWSTR
+    #define wxLPSTR LPWSTR
+    #define wxLPWNDCLASS LPWNDCLASSW
+    #define wxDOCINFO DOCINFOW
+    #define wxLPFINDREPLACE LPFINDREPLACEW
```

Comment:
Minor, but these could be `typedef`s (or even `using`) if we try to minimize the use of preprocessor (possibly not really helpful in this header...).